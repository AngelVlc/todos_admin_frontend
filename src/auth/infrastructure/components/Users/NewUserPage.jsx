import React from "react";
import { UserForm } from "./UserForm";
import { Breadcrumb } from "../../../../shared/infrastructure/components/Breadcrumb/Breadcrumb";

export const NewUserPage = () => {
  return (
    <div className="container">
      <Breadcrumb
        items={[
          { url: "/users", text: "Users" },
          { url: "/users/new", text: "New" },
        ]}
      />
      <h3 className="title">New user</h3>
      <UserForm />
    </div>
  );
};
