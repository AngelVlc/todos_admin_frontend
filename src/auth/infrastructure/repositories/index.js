export * from './RefreshTokensRepository';
export * from './UsersRepository';
