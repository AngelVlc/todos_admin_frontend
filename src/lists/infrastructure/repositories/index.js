export * from "./ListsRepository";
export * from "./ListItemsRepository";
