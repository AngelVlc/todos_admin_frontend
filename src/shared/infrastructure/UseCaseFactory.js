export class UseCaseFactory {
  get(useCase) {
    return useCase.create();
  }
}
