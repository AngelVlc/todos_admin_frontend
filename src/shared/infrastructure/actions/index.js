export * from './requests';
export * from './login';
