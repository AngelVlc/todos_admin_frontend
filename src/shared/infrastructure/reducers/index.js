export * from './loginReducer';
export * from './requestsReducer';
