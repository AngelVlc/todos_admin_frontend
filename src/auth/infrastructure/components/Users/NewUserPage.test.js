import { render, cleanup } from "@testing-library/react";
import { NewUserPage } from "./NewUserPage";
import { AppContext } from "../../../../shared/infrastructure/contexts";
import { MemoryRouter, Route } from "react-router-dom";

const mockHistoryPush = jest.fn();

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useHistory: () => ({
    push: mockHistoryPush,
  }),
}));

const renderWithContextAndRouterForNewUser = (component) => {
  const context = { auth: { info: {} } };
  return {
    ...render(
      <AppContext.Provider value={context}>
        <MemoryRouter initialEntries={[`/users/new`]}>
          <Route path="/users/new">{component}</Route>
        </MemoryRouter>
      </AppContext.Provider>
    ),
  };
};

afterEach(cleanup);

it("should match the snapshot for a new user", async () => {
  const { asFragment } = renderWithContextAndRouterForNewUser(<NewUserPage />);
  expect(asFragment(<NewUserPage />)).toMatchSnapshot();
});
