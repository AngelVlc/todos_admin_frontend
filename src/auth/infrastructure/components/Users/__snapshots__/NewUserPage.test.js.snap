// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot for a new user 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <nav
      aria-label="breadcrumbs"
      class="breadcrumb"
    >
      <ul>
        <li>
          <a
            href="/"
          >
            Home
          </a>
        </li>
        <li>
          <a
            href="/users"
          >
            Users
          </a>
        </li>
        <li
          class="is-active"
        >
          <a
            aria-current="page"
            href="/users/new"
          >
            New
          </a>
        </li>
      </ul>
    </nav>
    <h3
      class="title"
    >
      New user
    </h3>
    <form
      action="#"
    >
      <div
        class="field"
      >
        <label
          class="label"
          for="name"
        >
          Name
        </label>
        <div
          class="control"
        >
          <input
            data-testid="name"
            name="name"
            type="text"
            value=""
          />
        </div>
        <p
          class="help is-danger"
          data-testid="userNameErrors"
        />
      </div>
      <div
        class="field"
      >
        <label
          class="label"
          for="password"
        >
          Password
        </label>
        <div
          class="control"
        >
          <input
            data-testid="password"
            name="password"
            type="password"
            value=""
          />
        </div>
      </div>
      <div
        class="field"
      >
        <label
          class="label"
          for="confirmPassword"
        >
          Confirm Password
        </label>
        <div
          class="control"
        >
          <input
            data-testid="confirmPassword"
            name="confirmPassword"
            type="password"
            value=""
          />
        </div>
      </div>
      <div
        class="field"
      >
        <label
          class="label"
          for="isAdmin"
        >
          Is Admin
        </label>
        <div
          class="control"
        >
          <input
            data-testid="isAdmin"
            name="isAdmin"
            type="checkbox"
            value="false"
          />
        </div>
      </div>
      <div
        class="field is-grouped"
      >
        <div
          class="control"
        >
          <button
            class="button"
            data-testid="submit"
            type="submit"
          >
            CREATE
          </button>
        </div>
        <div
          class="control"
        >
          <button
            class="button"
            data-testid="cancel"
            type="button"
          >
            CANCEL
          </button>
        </div>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
